include/mitsuba/render/shape.h

```diff
@@ -327,6 +329,12 @@ class MTS_EXPORT_RENDER Shape : public Object {
     /// Is this shape a triangle mesh?
     bool is_mesh() const { return m_mesh; }
```

Comment:
The `Shape` class has racked up quite a few of these fields that add  storage to every Mitsuba shape. We already have this information via the `m_class` member. I would be in favor of removing all of those fields and just writing `m_class->derives_from(MTS_CLASS(Mesh))` and `m_class->name() == "Instance"` and `m_class->name() == "ShapeGroup"` in the other two case. AFAIK none of these functions are actually called during rendering (just scene initialization), so this should be fine. (but I could be wrong)